### ISSUE TITLE
modifying the state after Forking doesn't work

### ISSUE THREAD

**Issue by zgorizzo69** [nomiclabs/hardhat#1088]
**description**
I am getting a balance of an ERC20 and transfer that whole balance to another address impersonating the sender's address.
The balance of the receiver is correct but the balance  of the sender doesn't change. Moreover if I try to send the whole balance twice it works.

**expected behavior:**
The balance of the sender should be 0 after sending the total balance. The transfer should revert when trying to transfer twice the whole balance of an account.

To reproduce 
```
$ git clone https://github.com/zgorizzo69/hardhat
$ cd hardhat && vi .secret.alchemy
// Enter your alchemy project ID
$ yarn && yarn test
```

**Comment by joyious** [comment]
Same here.

Seems this is newly introduced in 2.0.4, tried revert back to 2.0.3 and it works as expected.

**Comment by alcuadrado** [comment]
Hey, @zgorizzo69, that's the expected result/failure here? I just run those commands and the tests passed.

@joyious, can you help us reproduce this? Thanks

**Comment by joyious** [comment]
@alcuadrado, you can see in the case, it transfers the token balance twice, this should not happen.

https://github.com/zgorizzo69/hardhat/blob/2137b32dbc4b0e7dfbdf18f12025b1c9fca341f0/test/comp-balance-impersonate.ts#L42

https://github.com/zgorizzo69/hardhat/blob/2137b32dbc4b0e7dfbdf18f12025b1c9fca341f0/test/comp-balance-impersonate.ts#L58

**Comment by zgorizzo69** [comment]
@alcuadrado you should check the comment in the test The balance of the sender should be 0 after sending the total balance. The transfer should revert when trying to transfer twice the whole balance of an account.
I have already updated the test just make a pull

**Comment by alcuadrado** [comment]
Thanks, that's really useful. There's in fact a bug when writing a 0 to the contract storage. We are working on it and will release a fix later today.

**Comment by zgorizzo69** [comment]
> Thanks, that's really useful. There's in fact a bug when writing a 0 to the contract storage. We are working on it and will release a fix later today.

thank you for this cool tool ;)

**Comment by fvictorio** [comment]
We just published `hardhat@2.0.5` that should fix this. Could you give it a try?

**Comment by joyious** [comment]
It works like a charm. Thanks for your good work there!
